cpp/modmesh/toggle/RadixTree.cpp

```diff
@@ -72,6 +72,60 @@ void CallProfiler::print_profiling_result(const RadixTreeNode<CallerProfile> & n
     }
 }
 
+void CallProfiler::serialize(std::ostream & outstream) const
+{
+    outstream << R"({)";
+    outstream << R"("radix_tree": )";
+    serialize_radix_tree(outstream);
+    outstream << R"(})";
+}
+
+void CallProfiler::serialize_radix_tree(std::ostream & outstream) const
+{
+    outstream << R"({)";
```

Comment:
I think you should do the JSON serialization yourself. Consider using a library, such as RapidJson, to help you wrap the object into JSON format. 

Here is what you can do:
1. define a [json schema](https://json-schema.org/)
2. implement the data structure in RadpidJson (or any other library)
3. use the library to dump the json format

Reply:
> I think you should not do the JSON serialization yourself. Consider using a library, such as RapidJson, to help you wrap the object into JSON format.
> 
> Here is what you can do:
> 
> 1. define a [json schema](https://json-schema.org/)
> 2. implement the data structure in RadpidJson (or any other library)
> 3. use the library to dump the json format

Does it mean that I need to install RapidJson library in [.github](https://github.com/solvcon/modmesh/tree/bc9e26f07585d74b2fb4d96c194ff209e7c78996/.github)/[workflows](https://github.com/solvcon/modmesh/tree/bc9e26f07585d74b2fb4d96c194ff209e7c78996/.github/workflows) ?

Reply:
you need to install RadpidJson in CMake, so you can use it in cpp files.
You can do a study first before opting in RapidJson, perhaps  there is a more suitable library for us.

Reply:
I like @ThreeMonth03 's current approach.

Developing the JSON writing code provides more freedom than using a thirdparty.  A thirdparty introduces additional dependency and increases the complexity of the building system.  JSON is easy, and writing a parser should be fun.

I do not see a good reason to use a thirdparty JSON library.

Reply:
See more discussion here: https://discord.com/channels/730297880140578906/1234815972809117778

Reply:
Per the discussion in Discord, I don't think we need a library for JSON, but if we need one, the current implementation is not general, we should have a JSON parser class instead.

Reply:
> Per the discussion in Discord, I don't think we need a library for JSON, but if we need one, the current implementation is not general, we should have a JSON parser class instead.

Does it mean that I should survey and implement a general json parser class?